tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc

```diff
@@ -279,6 +275,44 @@ Type ConvertType(Type t) {
 
 }  // namespace
 
+// Converts the blocks signature and allocate buffers for the block arguments.
+void BlockArgsConversionAndAllocation(FuncOp& funcOp,
+                                      ConversionPatternRewriter& rewriter) {
+  for (auto& block : funcOp.getBlocks()) {
+    if (auto op = dyn_cast<CondBranchOp>(block.getTerminator())) {
+      op.emitOpError() << "Conditional branch operation is not yet supported.";
+    }
+
+    // A buffer must be allocated for each block successors' argument within
+    // the current block. All branches' operands must be copied into and
+    // replaced with the new allocated buffer.
+    if (block.getNumSuccessors() > 0) {
+      auto terminatorOp = block.getTerminator();
+      auto loc = terminatorOp->getLoc();
+      rewriter.setInsertionPoint(terminatorOp);
+
+      for (auto operand : terminatorOp->getOperands()) {
+        auto t = operand.getType().dyn_cast<ShapedType>();
+        auto memrefType = MemRefType::get(t.getShape(), t.getElementType());
+        auto alloc = rewriter.create<AllocOp>(loc, memrefType);
+        alloc.setAttr(kTempBufferAttr, rewriter.getBoolAttr(true));
```

Comment:
@herhut: I did not find any usage of "temp" attribute.

I do not think it is used anywhere, let's get rid of tempbufferattr altogether.